skrub/tests/test_deduplicate.py

```diff
@@ -95,3 +98,67 @@ def test__create_spelling_correction(seed: int = 123) -> None:
             spelling_correction.values[clusters == n].astype("int")
             == counts[clusters == n].max()
         ).all()
+
+
+@skip_if_no_parallel
+def test_parallelism():
+    # Test that parallelism works
+    X = make_deduplication_data(examples=['black', 'white', 'red'], entries_per_example=[500, 500, 500], prob_mistake_per_letter=0.3)
+    y = deduplicate(X, n_jobs=None)
+    times = {}
+    jobs = [1, 4]
+
+    for n_jobs in jobs:
+        start = time.time()
+        y_parallel = deduplicate(X, n_jobs=n_jobs)
+        assert_array_equal(y, y_parallel)
+        end = time.time()
+        timer = end - start
+        times[n_jobs] = timer
+    print(times)
+    # Test if execution time is shorter:
+    for i in range(len(jobs)-1):
+        assert times[jobs[i+1]] < times[jobs[i]]
+
+    # Test with threading backend
+    with joblib.parallel_backend("threading"):
+        y_threading = deduplicate(X, n_jobs=1)
+    assert_array_equal(y, y_threading)
+
+
+DEFAULT_JOBLIB_BACKEND = joblib.parallel.get_active_backend()[0].__class__
+
+
+class DummyBackend(DEFAULT_JOBLIB_BACKEND):  # type: ignore
```

Comment:
Fun! Maybe we should add such a pattern in joblib and document it in a "how to test with joblib",

cc @tomMoral

Reply:
Yes, maybe adding a way to test that the `parallel_config` is used by downstream libraries would be nice.